weblate/trans/views/basic.py

```diff
@@ -231,6 +231,12 @@ def show_project_language(request, obj):
         user, project=project_object, language=language_object
     ).recent()
 
+    last_announcements = (
+        Change.objects.last_changes(user, project=None, language=language_object)
```

Comment:
Project filtering is missing here:

```suggestion
        Change.objects.last_changes(user, project=project_object, language=language_object)
```